csharp/src/Apache.Arrow/Ipc/ArrowStreamWriter.cs

```diff
@@ -204,7 +204,7 @@ private void CreateBuffers<T>(PrimitiveArray<T> array)
 
             private Buffer CreateBuffer(ArrowBuffer buffer)
             {
-                int offset = TotalLength;
+                int offset = checked((int)TotalLength);
```

Comment:
Both "offset" and the properties in the Buffer class would also need to be changed to long.